rbx_xml/src/serializer_core.rs

```diff
@@ -63,6 +63,25 @@ impl<W: Write> XmlEventWriter<W> {
         Ok(())
     }
 
+    /// Writes a `f32` value, but using Roblox's INF, -INF, and NAN values
+    #[inline]
+    pub fn write_characters_f32(&mut self, value: f32) -> Result<(), NewEncodeError> {
+        self.write_characters_f64(value as f64)
+    }
+
+    /// Writes a `f64` value, but using Roblox's INF, -INF, and NAN values
+    pub fn write_characters_f64(&mut self, value: f64) -> Result<(), NewEncodeError> {
+        if value == std::f64::INFINITY {
+            self.write_characters("INF")
+        } else if value == std::f64::NEG_INFINITY {
+            self.write_characters("-INF")
+        } else if value.is_nan() {
+            self.write_characters("NAN")
+        } else {
+            self.write_characters(value)
+        }
```

Comment:
This might be cleaner as a `match` statement. For the NaN case, we might be able to use a match guard like `x if x.is_nan() => { ... }`

Reply:
Didn't know that was a feature :eyes:

Reply:
```
warning: floating-point types cannot be used in patterns
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>
```